packages/block-library/src/template-part/edit/index.js

```diff
@@ -100,21 +110,14 @@ export default function TemplatePartEdit( {
 						/>
 					</div>
 				</BlockControls>
+			) }
+			{ isTemplateFile && (
 				<TemplatePartInnerBlocks
 					postId={ postId }
 					hasInnerBlocks={ innerBlocks.length > 0 }
 				/>
-			</BlockWrapper>
-		);
-	}
-	if ( ! initialSlug.current && ! initialTheme.current ) {
-		// Fresh new block.
-		return (
-			<BlockWrapper>
-				<TemplatePartPlaceholder setAttributes={ setAttributes } />
-			</BlockWrapper>
-		);
-	}
-	// Part of a template file, post ID not resolved yet.
-	return null;
+			) }
+			{ isUnresolvedTemplateFile && __( 'Loading…' ) }
```

Comment:
This looks like a good place to use [`<Spinner />`](https://github.com/WordPress/gutenberg/tree/master/packages/components/src/spinner).

Reply:
Alright :)